pkg/resourcemanager/config/env.go

```diff
@@ -68,6 +68,17 @@ func ParseEnvironment() error {
 	}
 	targetNamespaces = ParseStringListFromEnvironment("AZURE_TARGET_NAMESPACES")
 
+	operatorMode = OperatorMode(envy.Get("AZURE_OPERATOR_MODE", "both"))
+	switch operatorMode {
+	case OperatorModeBoth, OperatorModeWebhooks, OperatorModeWatchers:
+	default:
+		return errors.Errorf(`operator mode must be one of "both", "webhooks" or "watchers" but was %q`, operatorMode)
+	}
+
+	if operatorMode == OperatorModeWebhooks && len(targetNamespaces) > 0 {
```

Comment:
Doesn't this presuppose that there are only the two modes?

Would this be better written as
```suggestion
	if !operatorMode.IncludesWatchers() && len(targetNamespaces) > 0 {
```

and then change the error message accordingly to say: "mode must contain watchers to specify targetNamespaces"?